src/ffs/ffs.ts

```diff
@@ -0,0 +1,109 @@
+import pubTree, { PublicTree } from './pub/tree'
+import privTree, { PrivateTree } from './priv/tree'
+import pubNode, { PublicNode } from './pub/node'
+import { Tree, Node, Link } from './types'
+import { CID, FileContent } from '../ipfs'
+import pathUtil from './path'
+
+export class FFS {
+
+  root: PublicNode
+  pubTree: PublicTree
+  privTree: PrivateTree
+
+  constructor(root: PublicNode, pubTree: PublicTree, privTree: PrivateTree) {
+    this.root = root
+    this.pubTree = pubTree
+    this.privTree = privTree
+  }
+
+  static async empty(rootKeyName: string = 'filesystem-root'): Promise<FFS> {
+    const root = await pubNode.empty()
+    const pubTreeInstance = await pubTree.empty()
+    const privTreeInstance = await privTree.empty(rootKeyName)
+    return new FFS(root, pubTreeInstance, privTreeInstance)
+  }
+
+  // upgrade public IPFS folder to FileSystem
+  static async upgradePublicCID(cid: CID, rootKeyName: string = 'filesystem-root'): Promise<FFS> {
```

Comment:
@dholms Does this do the following?

```shell
# Suppose I have the CID `Qm111` of a __folder__ with the following contents
file_a.txt
file_b.pdf
```

If I then run `upgradePublicCID("Qm111")`, would I get following tree?

```shell
# New Tree
/public/file_a.txt
/public/file_b.pdf
```

Reply:
yes but it also creates an empty private folder as well. so you'd get 
```
root
  public
    file_a.txt
    file_b.pdf
  private
```